### DISCUSSION TITLE
一直报错 FATAL:  role "root" does not exist

### DISCUSSION THREAD

**Discussion by wxjun2018** [swuecho/chat#91] [Q&A] [answered]
web        | 218.82.136.139 - - [10/Apr/2023:09:13:46 +0000] "POST /api/chat_stream HTTP/1.1" 500 61 "http://1x:8084/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36" "-"
api        | 172.18.0.4 - - [10/Apr/2023:09:13:46 +0000] "POST /chat_stream HTTP/1.1" 500 61
chat-db-1  | 2023-04-10 17:13:46.927 CST [156] FATAL:  role "root" does not exist
api        | 2023/04/10 09:13:50 chatSessionUuid: 1fdf2591-898a-4af6-9a62-d0be5a00fead
api        | 2023/04/10 09:13:50 chatUuid: 1e5032dd-0c0c-45ea-a04f-d68c412b05f6
api        | 2023/04/10 09:13:50 newQuestion: 1+1等于多少
api        | 
api        | 172.18.0.4 - - [10/Apr/2023:09:13:50 +0000] "POST /chat_stream HTTP/1.1" 200 0
web        | 218.82.136.139 - - [10/Apr/2023:09:13:50 +0000] "POST /api/chat_stream HTTP/1.1" 200 0 "http://1x:8084/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36" "-"
chat-db-1  | 2023-04-10 17:13:52.070 CST [165] FATAL:  role "root" does not exist
chat-db-1  | 2023-04-10 17:13:57.215 CST [173] FATAL:  role "root" does not exist
chat-db-1  | 2023-04-10 17:14:02.357 CST [181] FATAL:  role "root" does not exist

**Comment by swuecho** [comment]
chatgpt 答案, 你看一下docker-compose 里面你的 PG_USER

----

This error message usually occurs when trying to access PostgreSQL database using the root user.

To fix this error, you need to create a new PostgreSQL user with administrative privileges and grant access to the database.

Here is an example code snippet to create a new user and grant administrative access:

```sql
CREATE USER new_user WITH PASSWORD 'password';

ALTER ROLE new_user CREATEDB;

GRANT ALL PRIVILEGES ON DATABASE my_database TO new_user;
```

In the above code, replace "new_user" with the desired username and "password" with the desired password. Also, replace "my_database" with the name of your PostgreSQL database.

Once you have created the new user and granted access, you can connect to the database using the new user credentials instead of root.

**Comment by swuecho** [comment]
https://github.com/swuecho/chat/actions/runs/4655410555/jobs/8238056235

看了下测试的log, 也有类似的问题, 貌似这个 root user 不影响. 确认一下你的 api key 是否正确, 有效.

贴一下你的 docker-compose file

**Comment by wxjun2018** [comment]
root user 不影响注册和登录。就是一直报比较烦人

```
version: '3'

services:
  web:
    container_name: web
    image: ghcr.io/swuecho/chat_frontend:latest # or use tag for better stability, e.g. v0.0.3
    ports:
      - 8084:80
    depends_on:
      - api
      - db
  api:
    container_name: api
    image: ghcr.io/swuecho/chat_backend:latest # or use tag for better stability, e.g.v0.0.3
    expose:
      - 8077
    ports:
      - 8077:8077
    environment:
      # at least one key is required.
      - OPENAI_API_KEY="sk-CCMeUnAYZhIfxaV3p1dLT3BlbmXXXXXXXXXXX" # do not change if you do not have openai api key
      - CLAUDE_API_KEY=thisisclaudekey # do not change if you do not have claude api key
      # api call in 10min
      # set this to zero if your server is in public network. only increase ratelimit in admin panel for trusted users.
      - OPENAI_RATELIMIT=100
      # DB config, set based on your db config if you don't use the db in docker-compose
      - PG_HOST=db
      - PG_DB=postgres
      - PG_USER=postgres
      - PG_PASS=thisisapassword
      - PG_PORT=5432
    depends_on:
      db:
        condition: service_healthy
  db:
    image: postgres:14
    restart: always
    environment:
      TZ: "Asia/Shanghai"
      PGTZ: "Asia/Shanghai"
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "thisisapassword"
    expose:
      - 5432
    ports:
      - "5432:5432"
    healthcheck:
      test:
        [
          "CMD-SHELL",
          "pg_isready",
          "-q",
          "-d",
          "postgres",
          "-U",
          "postgres"
        ]
      interval: 5s
      timeout: 5s
      retries: 5
```

**Comment by swuecho** [comment, marked as answer]
> - OPENAI_API_KEY="sk-CCMeUnAYZhIfxaV3p1dLT3BlbmXXXXXXXXXXX" # do not change if you do not have openai api key

- OPENAI_API_KEY=sk-CCMeUnAYZhIfxaV3p1dLT3BlbmXXXXXXXXXXX # do not change if you do not have openai api key

去掉 “

docker 这个地方比较坑

**Comment by swuecho** [comment]
https://github.com/swuecho/chat/pull/89

解决 FATAL: role "root" does not exist
